docs/reference-architectures/data/monitoring.md

```diff
@@ -0,0 +1,147 @@
+
+# Monitoring Apache Spark Structured Streaming queries in Azure Databricks
+
+Azure Databricks is based on Apache Spark, which includes a set of structured APIs for batch processing data using Datasets, DataFrames, and SQL. With Apache Spark 2.0, support was added for [Structured Streaming](https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html), a data stream processing API built upon Spark's batch processing APIs.
+
+The Structured Streaming API includes functionality to monitor active streams as data is processed by sending event information to an external metrics system. The external metrics system is registered with Spark Structured Streaming by attaching an implementation of a **StreamingQueryListener** object to the current Spark streaming session.
+
+This solution accelerator includes a **StreamingQueryListener** implementation, **LogAnalyticsStreamingQueryListener**, to send Structured Streaming event information to an Azure Log Analytics workspace. You first attach the **LogAnalyticsStreamingQueryListener** to the Spark session in your job application code. Next, the solution accelerator includes a Spark initialization script that copies the **LogAnalyticsStreamingQueryListener** implementation to each node in your Spark cluster. Finally, specify the connection string, workspace ID, and shared access signature (SAS) token for your Log Analytics workspace in the Spark configuration settings. When your job runs, metrics are sent to the specified Log Analytics workspace.
+
+# Use the solution accelerator
+
+This solution accelerator has the following prerequisites:
+
+* Clone, fork, or download the [GitHub repository](https://github.com/mspnp/spark-monitoring).
+
+* An Azure Databricks workspace. For instructions, see [get started with Azure Databricks.](https://docs.microsoft.com/azure/azure-databricks/quickstart-create-databricks-workspace-portal).
+* Install the [Azure Databricks CLI](https://docs.databricks.com/user-guide/dev-tools/databricks-cli.html#install-the-cli).
+  * An Azure Databricks personal access token is required to use the CLI. For instructions, see [token management](https://docs.azuredatabricks.net/api/latest/authentication.html#token-management).
+  * You can also use the Azure Databricks CLI from the Azure Cloud Shell. 
+
+* A Java IDE, with the following resources:
+  * JDK 1.8
+  * Scala SDK 2.11
+  * Maven 3.5.4
+
+To use the solution accelerator, import the Maven project configuration file, _pom.xml_, located in the **spark-monitoring** folder into your project. This will import four projects:
+
+* spark-listeners
+* spark-metrics
+* spark-monitoring
+* spark-jobs
+
+For this solution accelerator, you'll work with the **spark-listeners** project.
+
+## Develop your Apache Spark job application for Azure Databricks
+
+There are three requirements for using the **LogAnalyticsStreamingQueryListener** object in your Apache Spark job application code:
```

Comment:
What is the purpose of the job application? I know it is supposed to hook up You first attach the **LogAnalyticsStreamingQueryListener** to the Spark session. Other tasks?